.github/workflows/lint.yml

```diff
@@ -52,6 +52,24 @@ jobs:
         find src/go -type f -not -name 'zz*' -name '*.go' | xargs -n1 "$HOME/.local/bin/gofumpt" -w -lang=1.20
         git diff --exit-code
 
+    - name: go mod tidy (cluster-to-redpanda-migration)
+      working-directory: src/go/cluster-to-redpanda-migration
+      run: |
+        go mod tidy
+        git diff --exit-code -- go.mod go.sum
+
+    - name: go mod tidy (k8s)
+      working-directory: src/go/k8s
+      run: |
+        go mod tidy
+        git diff --exit-code -- go.mod go.sum
+
+    - name: go mod tidy (rpk)
+      working-directory: src/go/rpk
+      run: |
+        go mod tidy
+        git diff --exit-code -- go.mod go.sum
+
```

Comment:
I don't want to test all the unrelated packages every time. If something fails outside of the project I'm working on, I don't want to have to include an unrelated commit just to get CI to pass.

Reply:
this takes 1 minute to run and is going to help avoid the issue we're in right now. Since this is blocking the release, could we please merge as is and address later? We will address the issue of proper monorepo filters really soon